src/Raven.Server/Routing/RequestRouter.cs

```diff
@@ -223,18 +223,28 @@ public static bool TryGetClientVersion(HttpContext context, out Version version)
         public static void AssertClientVersion(HttpContext context, Exception innerException)
         {
             // client in this context could be also a follower sending a command to his leader.
-            if (TryGetClientVersion(context, out var clientVersion))
-            {
-                var currentServerVersion = RavenVersionAttribute.Instance;
+            if (TryGetClientVersion(context, out var clientVersion) == false) 
+                return;
 
-                if (currentServerVersion.MajorVersion != clientVersion.Major || currentServerVersion.BuildVersion < clientVersion.Revision || currentServerVersion.BuildVersion == ServerVersion.DevBuildNumber || (clientVersion.Revision >= 40 && clientVersion.Revision < 50))
-                {
-                    throw new ClientVersionMismatchException(
-                        $"Failed to make a request from a newer client with build version {clientVersion} to an older server with build version {RavenVersionAttribute.Instance.AssemblyVersion}.{Environment.NewLine}" +
-                        $"Upgrading this node might fix this issue.",
-                        innerException);
-                }
-            }
+            if(CheckClientVersionAndWrapException(clientVersion, ref innerException) == false)
+                throw innerException;
+        }
+
+        public static bool CheckClientVersionAndWrapException(Version clientVersion, ref Exception innerException)
+        {
+            var currentServerVersion = RavenVersionAttribute.Instance;
+
+            if (currentServerVersion.MajorVersion == clientVersion.Major &&
+                currentServerVersion.BuildVersion >= clientVersion.Revision &&
```

Comment:
Why are you comparing build version to revision here?

Reply:
There are two different classes.
The `currentServerVersion` is `RavenVersionAttribute`
And the `clientVersion` is System.Version.
So the `RavenVersionAttribute.BuildVersion` is equivalent to `System.Version.Revision`